### PR TITLE
Add PyPI publishing in GH workflow

### LINKED ISSUE #238: Release: Automatic release to PyPI
### What do you want to share with us?

Implement in CD the automatic release to PyPI (https://pypi.org/project/repository-service-tuf/)

I good reference is the work done in the python-tuf library.

### References

- https://github.com/theupdateframework/python-tuf/blob/develop/.github/workflows/cd.yml

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by KAUTH**
The process of publishing the package to PyPI will now be automated.
TODO: Replace secret's name with the repo's one for PyPI (in -> Security -> Secrets and variables -> Actions)

Note: If we want we can test this first before merging by releasing to TestPyPI (will need a new token with project scope).

Closes #238

### CODE REVIEW

**Comment by kairoaraujo** [comment]
Hi @KAUTH, thank you! It looks good.
I think adding the configuration for test.pypi.org before uploading it to the PyPI production repo is important.

**Comment by KAUTH** [comment]
> Hi @KAUTH, thank you! It looks good. I think adding the configuration for test.pypi.org before uploading it to the PyPI production repo is important.

Agreed, I added it :smiley: The only thing left is for a repo and PyPI admin to match `PYPI_API_TOKEN` and  `TEST_PYPI_API_TOKEN` secrets.

**Comment by kairoaraujo** [comment]
> > Hi @KAUTH, thank you! It looks good. I think adding the configuration for test.pypi.org before uploading it to the PyPI production repo is important.
> 
> Agreed, I added it 😃 The only thing left is for a repo and PyPI admin to match `PYPI_API_TOKEN` and `TEST_PYPI_API_TOKEN` secrets.

I will work on this part ;)

**Comment by KAUTH** [comment]
Let me know if we need anything from my side before closing this PR :smiley:

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -79,4 +82,18 @@ jobs:
               repo: context.repo.repo,
               release_id: '${{ needs.build.outputs.release_id }}',
               name: '${{ github.ref_name }}',
-            })
\ No newline at end of file
+            })
+
+      - name: Publish distribution 📦 to Test PyPI
+        if: github.repository == 'vmware/repository-service-tuf-cli'
```
Comment (kairoaraujo) [inline .github/workflows/cd.yml:88]:
```suggestion
        if: github.repository == 'repository-service-tuf/repository-service-tuf-cli'
```

File: .github/workflows/cd.yml [hunk 2]
```diff
@@ -79,4 +82,18 @@ jobs:
...
+        with:
+          password: ${{ secrets.TEST_PYPI_API_TOKEN }}
+          repository-url: https://test.pypi.org/legacy/
+
+      - name: Publish binary wheel and source tarball 📦 on PyPI
+        # Only attempt PyPI upload in upstream repository
+        if: github.repository == 'vmware/repository-service-tuf-cli'
+        uses: pypa/gh-action-pypi-publish@0bf742be3ebe032c25dd15117957dc15d0cfc38d
+        with:
+          password: ${{ secrets.PYPI_API_TOKEN }}
```
Comment (kairoaraujo) [inline .github/workflows/cd.yml:99]:
```suggestion
```
We can use [trusted publishing](https://github.com/marketplace/actions/pypi-publish#trusted-publishing) with this action.
I already configured it on PyPI.

**Review by kairoaraujo — CHANGES_REQUESTED** [review]

File: .github/workflows/cd.yml [hunk 3]
```diff
@@ -79,4 +82,18 @@ jobs:
...
               name: '${{ github.ref_name }}',
-            })
\ No newline at end of file
+            })
+
+      - name: Publish distribution 📦 to Test PyPI
+        if: github.repository == 'vmware/repository-service-tuf-cli'
+        uses: pypa/gh-action-pypi-publish@0bf742be3ebe032c25dd15117957dc15d0cfc38d
+        with:
+          password: ${{ secrets.TEST_PYPI_API_TOKEN }}
```
[the remark this answers is not in the document]
Comment (KAUTH) [reply .github/workflows/cd.yml]: I believe I adjusted it, I am not sure if the extra env variables are needed at each publish step but I added them following the `gh-action-pypi-publish` docs.

**Review by kairoaraujo — APPROVED** [review]

**Pull request merged**